src/services/paras/ParasService.ts

```diff
@@ -437,7 +463,26 @@ export class ParasService extends AbstractService {
 			return null;
 		}
 
-		return now.sub(new BN(beginEnd)).iadd(BN_ONE);
+		/**
+		 * The length of each sample to take during the ending period.
+		 *
+		 * A sample can be represented by `afterEarlyEnd` / `sampleLength`.
+		 * When we are in the endingPeriod, the offset is represented by:
+		 * `sample - 1`.
+		 */
+		const sampleLength = historicApi.consts.auctions
+			.sampleLength as BlockNumber;
+
+		switch (phase) {
+			case 'startPeriod':
+				return BN_ZERO;
+			case 'endPeriod':
+				return afterEarlyEnd.div(sampleLength).sub(BN_ONE);
+			case 'vrfDelay':
+				return BN_ZERO;
```

Comment:
Whats the justification for returning zero in this case?

Reply:
I would think it would be more intuitive to return the offset into the last ending sample. In other words if we are in the vrf delay phase, we display who was winning at the last block of the auction

Reply:
So I am actually split between returning null in a vrfDelay phase (no winning information), and returning zero for an offset. 

The polkadot code doesn't return a zero, but instead errors when retrieving the offset during a `vrfDelay` phase, but winning data is still available during the vrfDelay when querying with zero via polkadot-js. Therefore I had it in. Whether it's right to do I am not totally sure tbh.

Probably safer to just result to the default `return null` in the case of `vrfDelay`?

Reply:
In polkadot, during the vrf delay there is not really notion of "currently winning" because we are waiting for the offset into the ending period. As a user of this API though it makes the most sense to either return:

1)  who was winning for the last ending sample, as this makes the most sense when query in order. For example, say during the last sample para 5 was winning, but during first sample para 0 was winning. If we are returning 5 as winning for all the queries during the last sample, but then all of a sudden start returning 0 as winning during the VRF delay it is confusing

2) not return any winners, and clearly add to the docs they should query the last block of the ending period to see who was winning at the end of the election

> The polkadot code doesn't return a zero, but instead errors when retrieving the offset during a vrfDelay 

This is because you can't bid in an auction that has ended (the linked code is for `handle_bid`) (sorry if I am just pointing out the obvious but wanted to clarify :P)

Reply:
The more I think about it, the more I think option 2 is "more correct" in terms of a consistent API

Reply:
> we display who was winning at the last block of the auction

Ahh okay, yea I can see that making sense. 

Basically instead of calculating the `afterEarlyEnd` via `now.sub(beginEnd)` it would be `(finishEnd - 1).sub(beginEnd)` I believe the minus one because bid arent accepted on the last block. (Ill have to double check).

Reply:
Weird how it's not updating the comments as were both typing haha!

> The more I think about it, the more I think option 2 is "more correct" in terms of a consistent API

Cool, yea at first I figured this was the best way to go about it too. So for now as you mentioned, ill actually just remove the whole `vrfDelay` check and have it default to null. Ill write some inline docs explaining how we handle the vrfDelay, then also update the user docs, explaining they should use the finishEnd as a block reference to query the winner. 

Also thanks for the explanation above super helpful.